### PR TITLE
OCPBUGS-4323: Move route/v1 defaulting from openshift-apiserver for reuse.

### PR DESCRIPTION
**Pull request by benluddy**

### CODE REVIEW

**Comment by benluddy** [comment]
Proof PR of openshift-apiserver using this: https://github.com/openshift/openshift-apiserver/pull/334.

**Review by sanchezl — APPROVED** [review]
/lgtm

File: Makefile [hunk 1]
```diff
@@ -8,11 +8,28 @@ GO_BUILD_PACKAGES :=$(GO_PACKAGES)
 # Do not auto-expand packages for libraries or it would compile them separately
 GO_BUILD_PACKAGES_EXPANDED :=$(GO_BUILD_PACKAGES)
 
+DEFAULTER_GEN ?= k8s.io/code-generator/cmd/defaulter-gen@v0.25.0
```
Comment (deads2k) [inline Makefile:11]: We don't want generated code here.

File: pkg/route/defaulting/defaults.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+import (
+	v1 "github.com/openshift/api/route/v1"
+)
+
+// If adding or changing route defaults, updates may be required to
+// pkg/router/controller/controller.go to ensure the routes generated from
+// ingress resources will match routes created via the api.
+
+func SetDefaults_RouteSpec(obj *v1.RouteSpec) {
```
Comment (deads2k) [inline pkg/route/defaulting/defaults.go:11]: You can have these if you want them.

**Comment by benluddy** [comment]
/hold

For presubmit results on the updated proof PR.

**Comment by benluddy** [comment]
All presubmits passed.

/hold cancel

**Pull request merged**